src/pudl/transform/epacems.py

```diff
@@ -18,16 +18,79 @@
 ###############################################################################
 
 
-def fix_up_dates(df: pd.DataFrame, plant_utc_offset: pd.DataFrame) -> pd.DataFrame:
-    """Fix the dates for the CEMS data.
+def harmonize_eia_epa_orispl(
+    df: pd.DataFrame,
+    pudl_engine: sa.engine.Engine,
+) -> pd.DataFrame:
+    """Harmonize the ORISPL code to match the EIA data.
+
+    The EIA plant IDs and CEMS ORISPL codes almost match, but not quite. EPA has
+    compiled a crosswalk that maps one set of IDs to the other. The crosswalk is
+    integrated into the PUDL db.
+
+    EIA IDs are more correct so use the crosswalk to fix any erronious EPA IDs and get
+    rid of that column to avoid confusion.
+
+    https://github.com/USEPA/camd-eia-crosswalk
+
+    Note that this transformation needs to be run *before* convert_to_utc, because
+    convert_to_utc uses the plant ID to look up timezones.
+
+    Args:
+        pudl_engine: SQLAlchemy connection engine for connecting to an existing PUDL DB.
+            This is used to access the crosswalk file for conversion. The crosswalk must
+            be processed prior to running this function or it won't work.
+        df: A CEMS hourly dataframe for one year-month-state.
+
+    Returns:
+        The same data, with the ORISPL plant codes corrected to match the EIA plant IDs.
```

Comment:
are the fixed or are the EIA codes just merged in? I'm not seeing replacement below, just want I think is a merge that adds the `plant_id_eia` column

Reply:
Yes, so previously the cems data had a column called `plant_id_eia`, but I changed it to `plant_id_epa` in the extract step because it was EPA's slightly inaccurate guess as to what the ORISPL code was. Most of the `plant_id_eia` values in CEMS were correct, but not all of them. The crosswalk helps fix the few cases where the EPA's ORISPL codes are bad by mapping the EPA's antiquated orispl plant code to the EIA's correct orispl code. 

This function merges the crosswalk with the cems data thus adding the official `plant_id_eia` column to CEMS. the `plant_id_combined` combines the `plant_id_eia` and `plant_id_epa` in cases where there is no `plant_id_eia` value (i.e., this plant isn't in the crosswalk yet). Because the `plant_id_epa` is almost always correct this feels reasonable!